docs/examples/gluster-s3-storage-template/README.md

```diff
@@ -2,111 +2,132 @@
 
 ## Prerequisites
 
-  OpenShift setup is up with master and nodes ready.
+OpenShift setup is up with master and nodes ready.
 
-  cns-deploy tool is ran and heketi service is ready
+cns-deploy tool is ran and heketi service is ready.
 
+## First create a storageclass
 
+We need to create a storageclass which provides access to glusterfs provisioner.
 
+Before creating storageclass, secret needs to be created and labelled. Please note, this step is optional.
 
-## First create a storageclass
+Create the secret:
 
-Create a GlusterFS StorageClass as below, replacing the `rest` parameters with your configuration:
+Replace `NAMESPACE` and `ADMIN_KEY` parameters with your configuration.
 
 ```
-oc create -f ./gluster-s3-storageclass.yaml
+oc create secret generic heketi-${NAMESPACE}-admin-secret --from-literal=key=${ADMIN_KEY} --type=kubernetes.io/glusterfs
 ```
 
-Available at 
-[gluster-s3-storageclass.yaml](./gluster-s3-storageclass.yaml)
+Label the secret:
 
-## Start glusters3 service using template
+Replace `NAMESPACE` parameter with your configuration.
 
 ```
-oc new-app gluster-s3-template.yaml  --param=S3_ACCOUNT=testvolume  --param=S3_USER=adminuser --param=S3_PASSWORD=itsmine --param=VOLUME_CAPACITY=2Gi
+oc label --overwrite secret heketi-${NAMESPACE}-admin-secret glusterfs=s3-heketi-${NAMESPACE}-admin-secret gluster-s3=heketi-${NAMESPACE}-admin-secret
 ```
 
-Note: adjust parameters according to your needs.
+Create a GlusterFS StorageClass as below, replacing the `HEKETI_URL` and `NAMESPACE` parameters with your configuration.
+Also, provide `STORAGE_CLASS` to give a storage class name.
+
+```
+sed  -e 's/${HEKETI_URL}/heketi-store-project1.cloudapps.mystorage.com/g'  -e 's/${STORAGE_CLASS}/gluster-s3-store/g'      -e  's/${NAMESPACE}/store-project1/g'   deploy/ocp-templates/gluster-s3-storageclass.yaml | oc create -f -
+```
 
+Available at 
+[gluster-s3-storageclass.yaml](../../../deploy/ocp-templates/gluster-s3-storageclass.yaml)
 
-If you wish to make use of a GlusterFS StorageClass other than `s3storageclass`, add another parameter of the form:
+## Create PVC
 
+Now, create PVC using the storage class. Replace `STORAGE_CLASS` with the above created one.
+Adjust `VOLUME_CAPACITY` as per your needs.
 
 ```
---param=STORAGE_CLASS=<your storage class name>
+sed -e 's/${VOLUME_CAPACITY}/2Gi/g'  -e  's/${STORAGE_CLASS}/gluster-s3-store/g'  deploy/ocp-templates/gluster-s3-pvcs.yaml | oc create -f -
+persistentvolumeclaim "gluster-s3-claim" created
 ```
 
+Available at
+[gluster-s3-pvcs.yaml](../../../deploy/ocp-templates/gluster-s3-pvcs.yaml)
+
+## Start glusters3 service using template
 
+Launch S3 storage service.
+
+Set S3_ACCOUNT name, S3_USER name, S3_PASSWORD.
+PVC and META_PVC are obtained from above step.
```

Comment:
Combine this with the previous sentence, and explain what the `S3_*` parameters are and that they can be whatever the admin wants to configure.

Also explain that PVC and META_PVC are the PVC names.